libraries/GCS_MAVLink/message_definitions/ardupilotmega.xml

```diff
@@ -555,8 +650,134 @@
+        <field name="serial_number_pt_3" type="uint32_t">Unit Serial Number Part 3 (incrementing serial number per month)</field>
+    </message>
+
+    <message name="GIMBAL_FACTORY_PARAMETERS_LOADED" id="207">
+        <description>
+            Sent by the gimbal after the factory parameters are successfully loaded, to inform the factory software that the load is complete
+        </description>
+        <field name="dummy" type="uint8_t">Dummy field because mavgen doesn't allow messages with no fields</field>
+    </message>
+
+    <message name="GIMBAL_ERASE_FIRMWARE_AND_CONFIG" id="208">
+        <description>
+            Commands the gimbal to erase its firmware image and flash configuration, leaving only the bootloader.  The gimbal will then reboot into the bootloader,
+            ready for the load of a new application firmware image.  Erasing the flash configuration will cause the gimbal to re-perform axis calibration when a
+            new firmware image is loaded, and will cause all tuning parameters to return to their factory defaults.  WARNING: sending this command will render a
+            gimbal inoperable until a new firmware image is loaded onto it.  For this reason, a particular "knock" value must be sent for the command to take effect.
+            Use this command at your own risk
+        </description>
+        <field name="target_system" type="uint8_t">System ID</field>
+        <field name="target_component" type="uint8_t">Component ID</field>
+        <field name="knock" type="uint32_t">Knock value to confirm this is a valid request</field>
+    </message>
+
+    <message name="GIMBAL_PERFORM_FACTORY_TESTS" id="209">
+        <description>
+            Command the gimbal to perform a series of factory tests.  Should not be needed by end users
+        </description>
+        <field name="target_system" type="uint8_t">System ID</field>
+        <field name="target_component" type="uint8_t">Component ID</field>
+    </message>
+
+    <message name="GIMBAL_REPORT_FACTORY_TESTS_PROGRESS" id="210">
+        <description>
+            Reports the current status of a section of a running factory test
+        </description>
+        <field name="test" type="uint8_t" enum="FACTORY_TEST">Which factory test is currently running</field>
+        <field name="test_section" type="uint8_t">Which section of the test is currently running.  The meaning of this is test-dependent</field>
+        <field name="test_section_progress" type="uint8_t">The progress of the current test section, 0x64=100%</field>
+        <field name="test_status" type="uint8_t">The status of the currently executing test section.  The meaning of this is test and section-dependent</field>
+    </message>
+
+    <!-- 211 to 214 RESERVED for more GIMBAL -->
+
+    <message name="GOPRO_POWER_ON" id="215">
+        <description>Instruct a HeroBus attached GoPro to power on</description>
```

Comment:
Gopro messages are closer to final now. But they can still be collapsed, and possible be using commandlong messages instead.